ports-of-call/static_vector.hpp

```diff
@@ -0,0 +1,436 @@
+
+  // back
+  // ---------------------------------------------------------------------------------------
+  PORTABLE_FUNCTION constexpr reference back() & { return ref(count_ - 1); }
+
+  PORTABLE_FUNCTION constexpr const_reference back() const & { return ref(count_ - 1); }
+
+  PORTABLE_FUNCTION constexpr Element &&back() && { return ref(count_ - 1); }
+
+  PORTABLE_FUNCTION constexpr Element const &&back() const && { return ref(count_ - 1); }
+
+  // data
+  // ---------------------------------------------------------------------------------------
+  PORTABLE_FUNCTION constexpr pointer data() { return ptr(); }
+
+  PORTABLE_FUNCTION constexpr const_pointer data() const { return ptr(); }
+
+  // begin
+  // --------------------------------------------------------------------------------------
+  PORTABLE_FUNCTION constexpr iterator begin() { return {ptr()}; }
+
+  PORTABLE_FUNCTION constexpr const_iterator begin() const { return {ptr()}; }
+
+  PORTABLE_FUNCTION constexpr const_iterator cbegin() const { return {ptr()}; }
+
+  // end
+  // ----------------------------------------------------------------------------------------
+  PORTABLE_FUNCTION constexpr iterator end() { return next(begin(), count_); }
+
+  PORTABLE_FUNCTION constexpr const_iterator end() const { return next(begin(), count_); }
+
+  PORTABLE_FUNCTION constexpr const_iterator cend() const {
+    return next(cbegin(), count_);
+  }
+
+  // empty
+  // --------------------------------------------------------------------------------------
+  PORTABLE_FUNCTION constexpr bool empty() const { return count_ == 0; }
+
+  // size
+  // ---------------------------------------------------------------------------------------
+  PORTABLE_FUNCTION constexpr size_type size() const { return count_; }
+
+  // max_size
+  // -----------------------------------------------------------------------------------
+  PORTABLE_FUNCTION constexpr size_type max_size() const { return N; }
+
+  // capacity
+  // -----------------------------------------------------------------------------------
+  PORTABLE_FUNCTION constexpr size_type capacity() const { return max_size(); }
+
+  // clear
+  // --------------------------------------------------------------------------------------
+  PORTABLE_FUNCTION constexpr void clear() noexcept {
+    for (size_type i = 0; i < count_; ++i) {
+      ptr(i)->~value_type();
+    }
+    count_ = 0;
+  }
+
+  // insert
+  // -------------------------------------------------------------------------------------
+  // TODO: In order to simplify, I removed all versions of insert for now.
+
+  // emplace
+  // ------------------------------------------------------------------------------------
+  // TODO: In order to simplify, I removed all versions of emplace for now.
+
+  // erase
+  // --------------------------------------------------------------------------------------
+  // TODO: In order to simplify, I removed all versions of erase for now.
```

Comment:
I'm not convinced these are needed ever. There's no reason we have to be 100% "standards compliant."